app/custom-oauth/server/custom_oauth_server.js

```diff
@@ -212,10 +214,41 @@ export class CustomOAuth {
 				},
 			};
 
+			callbacks.add('afterValidateLogin', (login) => {
```

Comment:
I know this is WIP.  But make sure you add this somewhere else.  :) since this section *only* pertains to custom oauth not a good place.  

Adding a nextcloud oauth package (this will guarantee the service name) and then adding this call back there would be better choice.

Reply:
@geekgonecrazy Thank you for your suggestion :)
Should I create a new package `app/nextcloud` which will only have one file with the login callback code or should I move this code to `app/webdav/server` as it is related to auto inserting a webdav server?
To give an overview of what is happening: I am auto inserting nextcloud webdav server (found in the upload files button) whenever a user logs in to RC with the nextcloud oauth. Presently, the user has to enter his credentials to add any server.

Reply:
As you have it currently there is no oauth provider in Rocket.chat for nextcloud.  So you are making the assumption that it will be created by the admin and that it’s name will be exactly nextcloud.

Instead you should copy the gitlab or any of the other oauth packages and make a nextcloud oauth provider.

Then inside there you can add this hook.  That way it’s toed directly to the provider that you know 100% what the name will be.  It also makes for a better bit of UX

Reply:
@geekgonecrazy I will do as you mentioned. Thanks!